### PR TITLE
Resolver Height (get by spent script)

### PR DESCRIPTION
**Pull request by crisdut**
This PR intend implement the `ResolverHeight` with combination of two electrum methods: **get_transaction**  and **batch_script_get_history**.

### CODE REVIEW

**Review by dr-orlovsky — APPROVED** [review]
ACK aebfe81b1a21532831deb5057ef8853ab9c68468

That is clever! Have been looking to solve electrum inability to provide tx height for long time, and never had this idea!

File: src/wallet.rs [hunk 1]
```diff
@@ -180,10 +178,42 @@ mod _electrum {
...
+                .iter()
+                .map(|out| out.script_pubkey.as_script())
+                .collect();
+
+            let mut hists = vec![];
+            self.0
+                .batch_script_get_history(scripts)
+                .map_err(|err| match err {
+                    Error::Message(_) | Error::InvalidResponse(_) => {
+                        TxResolverError::Other(txid, String::new())
```
[the remark this answers is not in the document]
Comment (crisdut) [reply src/wallet.rs:202]: I think this is a error typing =) 

I will fix!

**Review by dr-orlovsky — COMMENTED** [review]
Can you pls provide than a `impl From<electrum_client::Error> for Error` not to duplicate the code and avoid such errors?

**Comment by crisdut** [comment]
> Can you pls provide than a `impl From<electrum_client::Error> for Error` not to duplicate the code and avoid such errors?

I think is not possible, because we use external information to construct the error message.

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/RGB-WG/rgb/commit/35079ef9a64bb25219fb68e848d1c9778436213d

**Pull request merged**